lib/dest/write-contents/index.js

```diff
@@ -12,6 +12,11 @@ var fo = require('../../file-operations');
 function writeContents(optResolver) {
 
   function writeFile(file, enc, callback) {
+    // Write it as a symlink
+    if (file.symlink) {
```

Comment:
I wasn't sure here if the criterium should be existence of `file.symlink` and/or `file.isSymbolic()`. Ideally, if vinyl-fs is consistent with adding the `symlink` property, it should not matter I guess. But as things stand, the `file.symlink` test only catches the vinyls as produced by `src()` with falsy `resolveSymlinks` option (for this file), I believe.